frostsnap_coordinator/src/frostsnap_persist.rs

```diff
@@ -0,0 +1,210 @@
+            }
+
+            let message = row.get::<_, BincodeWrapper<coordinator::Mutation>>(0)?.0;
+
+            Ok(message)
+        })?;
+
+        for mutation in row_iter {
+            let mutation = mutation?;
+            coordinator.apply_mutation(&mutation);
+        }
+
+        Ok(coordinator)
+    }
+
+    fn persist_update(conn: &mut rusqlite::Connection, update: Self::Update) -> anyhow::Result<()> {
+        for mutation in update {
+            conn.execute(
+                "INSERT INTO fs_core_messages (message, version) VALUES (?1, 0)",
+                params![BincodeWrapper(mutation)],
+            )?;
+        }
+        Ok(())
+    }
+}
+
+impl TakeStaged<VecDeque<coordinator::Mutation>> for FrostCoordinator {
+    fn take_staged_update(&mut self) -> Option<VecDeque<coordinator::Mutation>> {
+        let mutations = self.take_staged_mutations();
+        if mutations.is_empty() {
+            None
+        } else {
+            Some(mutations)
+        }
+    }
+}
+
+impl Persist<rusqlite::Connection> for Option<SigningSessionState> {
+    type Update = Self;
+    type InitParams = ();
+
+    fn initialize(
+        conn: &mut rusqlite::Connection,
+        _params: Self::InitParams,
+    ) -> anyhow::Result<Self> {
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS fs_signing_session_state (
+            state BLOB
+        )",
+            [],
+        )?;
+
+        let signing_session_state =
+            conn.query_row("SELECT state FROM fs_signing_session_state", [], |row| {
+                Ok(row.get::<_, BincodeWrapper<SigningSessionState>>(0)?.0)
+            });
+
+        let state = match signing_session_state {
+            Ok(signing_session_state) => Some(signing_session_state),
+            Err(rusqlite::Error::QueryReturnedNoRows) => None,
+            Err(e) => return Err(e.into()),
+        };
+        Ok(state)
+    }
+
+    fn persist_update(conn: &mut rusqlite::Connection, update: Self::Update) -> anyhow::Result<()> {
+        match update {
+            Some(signing_session_state) => {
+                conn.execute(
+                    "INSERT INTO fs_signing_session_state (state) VALUES (?1)",
+                    params![BincodeWrapper(signing_session_state)],
+                )?;
+            }
+            None => {
+                conn.execute("DELETE FROM fs_signing_session_state", [])?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl TakeStaged<Option<SigningSessionState>> for Option<SigningSessionState> {
+    fn take_staged_update(&mut self) -> Option<Option<SigningSessionState>> {
+        Some(self.clone())
+    }
+}
+
+#[derive(Default)]
+pub struct DeviceNames {
+    names: HashMap<DeviceId, String>,
+    mutations: VecDeque<(DeviceId, String)>,
+}
+
+impl DeviceNames {
+    pub fn insert(&mut self, device_id: DeviceId, name: String) {
+        self.names.insert(device_id, name.clone());
+        self.mutations.push_back((device_id, name));
```

Comment:
Should only add a mutation if the name has changed

Reply:
fixed: 5976023668cc920a94e36a3ac9afc2d04aebb10a